src/main/java/com/book/config/SpringSecurityWebAppConfig.java

```diff
@@ -48,8 +48,9 @@ public PasswordEncoder passwordEncoder() {
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
         http.csrf().disable().cors().and().authorizeHttpRequests()
-                .requestMatchers("/register").permitAll()
-                .requestMatchers("/login").permitAll()
+                .requestMatchers("/register/**").permitAll()
+                .requestMatchers("/login/**").permitAll()
+                .requestMatchers("/product/**").permitAll()
```

Comment:
why does `/product` also bypass security?

Reply:
later i will change it to only getting listing of products can bypass security so that anonymous users able to check products